bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -1121,11 +1134,28 @@ public void run()
             }
           }
         }
-      } catch (Exception e) {
-        logger.error("{}", DataList.this, e);
+      } catch (RuntimeException e) {
+        logger.warn("{}", this, e);
+      } finally {
+        logger.debug("{} exiting run", this);
       }
     }
+
+    @Override
+    public String toString()
+    {
+      return new ToStringBuilder(this, SHORT_CLASS_NAME_STYLE).append(DataList.this)
```

Comment:
How about using the same format for the future as well instead of creating the string with '@' separator